### ISSUE TITLE
3PG mix output sum

### ISSUE THREAD

**Issue by trotsiuk** [checcomi/threePGN-package#20]
@checcomi I had run the test for the 3PGMix. If I understood correctly, in the output, sum of 3 layers (e.g. Basal area) shall give the same result as if I would run the single site version. However the output very different. E.g . the BasalArea sum gives 3 times higher level. 

```
firstRun <- r3pgn(siteData = data_site[1,],
  climate = data_climate,
  parameters = data_param[,2],
  outputs = c(1:5, 10:12, 26:27))


data_siteMix$minAsw <- 30
firstRunMix <- r3pgnMix(siteData = data_siteMix,
  climate = data_climate[,,1],
  parameters = data_paramMix,
  outputs = c(1:5, 10:12, 26:27))


i <- 4
plot(firstRunMix$output[,i,1] +firstRunMix$output[,i,2]+firstRunMix$output[,i,3])
points(firstRun$output[,i,1],pch=20,col=2)
```
![rplot](https://user-images.githubusercontent.com/15097779/44652910-821bbf80-a9ed-11e8-8059-abb6c1f34329.png)

The output of the `r3pgn` is rather similar to one layer output of `r3pgnMix` 

```
plot(firstRunMix$output[,i,1] )
points(firstRun$output[,i,1],pch=20,col=2)
```

![rplot01](https://user-images.githubusercontent.com/15097779/44652899-7c25de80-a9ed-11e8-9526-4be8b27693c0.png)

Furthermore, now I see that the original `r3pgn` dive the negative values (red points) for the Basal area, at the beginning, meaning there is some issue in calculations.

**Comment by checcomi** [comment]
@trotsiuk , thanks for testing this. I also noticed it and my interpretation was that there is not competition for light implemented at this stage.
So probably you should divide the solar radiation available for each layer by the number of layer (in this case by 3). 
Notice that this is just my interpretation.

**Comment by trotsiuk** [comment]
@checcomi 
If I understood correctly, the current loop is
```
do ii = 1, nmonths	
 do layer = 1, noOfLayers
  ...
 end do
end do
```

But from what I remember David was saying, is that it was not possible to run by layers, as the light distribution (not sure now) can be calculated only after the height (diameter) distribution.  Or something in that direction. I can't recall fully, but can maybe ask David,

in his code there is some notes about it, e.g. 
`'calculate avDBH and variables derived from it after removing the bias 'the volume is purposefully not calculated in getDiameterDistributions because volume should not be updated here 'but it will be changed prematurely due to the change in AvStemMass during the mortality calculations, and hence diameter.`

**Comment by checcomi** [comment]
Yes, now we have the layer loop that goes from the beginning to the end of model computations. next step will be to stop and restart the layer loop where needed.

**Comment by trotsiuk** [comment]
Ok, thank you! Would you have time also for it, or shall I try implement it myself.

**Comment by checcomi** [comment]
I'll code it and then you do the tests as you did for the layer loop

**Comment by checcomi** [comment]
@trotsiuk  
I reinitialized the loop in the code. The output should be exactly the same then before. 
now we need just to add the new functions in the right place.

**Comment by checcomi** [comment]
@trotsiuk can you check r3pgn.r file?

when I build the package and try to run the r3pgnMix function it doesn't recognize the function.
If i load the function manually then everything is working.
I think there is something wrong in the file but I cannot see it. see also the code below

 library(threePGN)
> firstRun <- r3pgn(siteData = data_site,
+                   climate = data_climate,
+                   parameters = data_param[,2],
+                   outputs = c(1:5, 10:12, 26:27))
> firstRunMix <- r3pgnMix(siteData = data_siteMix,
+                   climate = data_climate[,,1],
+                   parameters = data_paramMix,
+                   outputs = c(1:62))
Error in r3pgnMix(siteData = data_siteMix, climate = data_climate[, ,  : 
  could not find function "r3pgnMix"
> plot(firstRunMix$output[,3,1])
Error in plot(firstRunMix$output[, 3, 1]) : 
  object 'firstRunMix' not found
> r3pgnMix <- function(siteData,
+                      climate,
+                      thinning = NULL,
+                      parameters,
+                      outputs = c(1:5,10:12,26:27)){
+ 
+   nMonths <- dim(climate)[1]
+   noOfLayers <- nrow(siteData)
+   totThinning <- sum(siteData$nThinning)
+ 
+   if( all( is.null( thinning ) ) | totThinning == 0.) thinning = matrix(0,2,6)
+ 
+   nvariables = length(outputs)
+ 
+   y <- array(-999,dim=c(nMonths,nvariables,noOfLayers))
+ 
+   # if( length(unique(siteData[['climId']])) != nClimID ) stop("Provided climate data don't match with site data")
+ 
+   out <- .Fortran('threePGNmix1Site',
+                   output = as.array(y),
+                   nMonths = as.integer(nMonths),
+                   noOfLayers = as.integer(noOfLayers),
+                   nvariables = as.integer(nvariables),
+                   vars = as.integer(outputs),
+                   siteData = as.matrix(siteData),
+                   totThinning = as.integer(totThinning),
+                   thinning = as.matrix(thinning),
+                   weather = as.array(climate),
+                   parameters = as.numeric(parameters))
+ 
+   out$vars <- outNam[outputs]
+   class(out) = "r3pgOutMix"
+   return(out)
+ }
> outNam <- c("StandAge","StemNo","BasArea","StandVol","avDBH","MAI",
+             "SLA","CanCover","LAI","WF","WR","WS","WL",
+             "TotalW","AvStemMass","fracBB","fAge","fVPD","fT",
+             "fCalpha","fCg","fFrost","fSW","fNutr","PhysMod","GPP",
+             "NPP","RadInt","alphaC","epsilon","CVI","m","pR","pS",
+             "pF","pFS","gammaF","lossWF","lossWR","wSmax","gammaN",
+             "Mortality","supIrrig","RunOff","fRainInt","RainInt",
+             "CanCond","WUE","EvapTransp","Transp","ASW","NEP","Rhet",
+             "Yr_C","Yl_C","O_C","Yr_N","Yl_N","O_N")
> firstRunMix <- r3pgnMix(siteData = data_siteMix,
+                   climate = data_climate[,,1],
+                   parameters = data_paramMix,
+                   outputs = c(1:62))
> plot(firstRunMix$output[,3,1])

**Comment by trotsiuk** [comment]
Issue was due to missing `export` and `useDynLib` for the new function bb4d85c43809ad5b1fef77c8f1eaabab6845f06d

Now it shall work @checcomi
